### ISSUE TITLE
Update appveyor to use 2.4 & trunk, fix some test warnings

### ISSUE THREAD

**Issue by MSP-Greg** [sparklemotion/nokogiri#1672]
I fixed most of the test warning due to 'use assert_nil' and 'assert_send', then added ruby 24 and trunk to the appveyor build matrix.

[Test results](https://ci.appveyor.com/project/MSP-Greg/nokogiri) and the two commits, [Fix test warnings](https://github.com/MSP-Greg/nokogiri/commit/e24cb0f216ed969dfc7aa61425885f0ce9e16e8a) and [appveyor update](https://github.com/MSP-Greg/nokogiri/commit/7f19016d9c16f0cb853e86dfd7d1e0915ed43b77).

1. Would you like a PR?
2. If so, one two, or only appveyor update?

Thanks for your work and updates to libxml and libxslt...

**Comment by flavorjones** [comment]
Hi, I apologize for not responding more promptly.

I'm considering removing appveyor, as it's not useful to me. We're running a rather [complex pipeline](https://ci.nokogiri.org/) which runs windows tests on the versions of Windows that I care about; and the tests run quickly and reliably.

Appveyor runs slowly and so the feedback loops are quite slow. What are your thoughts on it? Is Appveyor useful to you? Is maintaining the configuration worth the benefits?

**Comment by MSP-Greg** [comment]
Accepted, although not needed.

I've been an advocate of both Windows testing and testing with trunk.  Lars added the RubyInstaller2 (RI2) trunk build to the Appveyor CI; it would certainly be helpful if somewhere trunk was tested on windows.  Whether you use the RI2 build or ruby-loco, it doesn't matter to me.

People have many different opinions about testing with trunk.  My normal response is 'how would you like it if everyone waited a year before using your software?'

As an aside, I use trunk, and I've built from both the MSYS2 packages and the included xml/xslt files.  Never had any problems.  Also, thanks for Nokogiri, I use it to maintain my [doc site](https://msp-greg.github.io/).  The main repos update with a mouse click, but I should update Nokogiri (and others)...

I'm okay to close...

Thanks, Greg

**Comment by flavorjones** [comment]
I agree we should be testing against ruby trunk, on both Linux and Windows. I've created #1718 to drive that work, which shouldn't be too difficult.

Thanks again for talking about this with me!

**Comment by larskanis** [comment]
@flavorjones The main advantage of Appveyor (and travis-ci) is IMHO simplicity. There are only a few lines of code in the yml file, so that it's very easy to understand what's happening in the tests. I can also copy and paste some lines into my locale console to reproduce and fix things locally. Concourse is much more difficult in this respect. So for me Appveyor is valuable in the nokogiri project.

**Comment by MSP-Greg** [comment]
Mike & Lars,

Lars, you mention 'locally'.  Additionally, the fact that contributors can easily use Travis and Appveyor on their forks is also a benefit.

1. I'm not saying that Appveyor or Travis are perfect, but they are in common use.
2. I hope everyone (not directed towards the two of you) is mindful of not wasting free resources (Appveyor & Travis), but this is a 'mature repo' and the commits are infrequent.

**Comment by flavorjones** [comment]
@larskanis and @MSP-Greg 

Thanks for the comments. Unfortunately, Travis builds have been failing for months and nobody's felt the need to fix them. This tells me very clearly that they're of no value. If you have another way of interpreting that information, I'd like to understand your point of view.

@larskanis if you'd like to maintain the `appveyor.yml` then I'm fine with that; however I can show you how to quickly run one-off Windows tests against the Concourse instance and I think you'll agree it's easy once you know how (and you have proper credentials for it, as a committer). Ping me in IRC if you want to chat about it.

**Comment by larskanis** [comment]
> nobody's felt the need to fix them

In fact I tried to fix it [here](https://github.com/sparklemotion/nokogiri/compare/master...larskanis:travis-2.5) but the broken rubygems frozen string compat in rubygems-2.7.0..2.7.3 and [this rbx bug](https://github.com/bundler/bundler/issues/6163) stopped me, so that I postponed it, until these dependent issues are solved.

> you have proper credentials for it, as a committer

Yay, I already logged in with my github account, but didn't notice any additional possibilities.

Sometimes in the past the Concourse web app was very slow or I wasn't able to view the test outputs - but this seems to be fixed now.
